unified_planning/model/walkers/linear_checker.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2021 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import unified_planning as up
+import unified_planning.model.walkers as walkers
+from unified_planning.model.walkers.dag import DagWalker
+from unified_planning.model.operators import OperatorKind
+from typing import List, Optional, Set, Tuple
+
+
+class LinearChecker(DagWalker):
+    """Checks if the given expression is linear or not and returns the set of the fluents appearing in the expression.
+
+    Optionally takes a problem to consider static fluents as constants.
+
+    Important NOTE:
+    After the initialization, the problem given as input can not be modified
+    or the Simplifier behaviour is undefined."""
+
+    def __init__(self, problem: Optional["up.model.problem.Problem"] = None):
+        DagWalker.__init__(self)
+        self._static_fluents: Set["up.model.fluent.Fluent"] = (
+            problem.get_static_fluents() if problem is not None else set()
+        )
+
+    def get_fluents(
+        self, expression: "up.model.fnode.FNode"
+    ) -> Tuple[bool, Set["up.model.fnode.FNode"]]:
```

Comment:
Is it needed to return the set of fluents or would be enough to return another boolean flag saying if it contains fluents?